htp/htp_list.c

```diff
@@ -172,12 +172,15 @@ htp_status_t htp_list_array_replace(htp_list_array_t *l, size_t idx, void *e) {
 
     if (idx + 1 > l->current_size) return HTP_DECLINED;
 
-    size_t i = l->first;
-
-    while (idx--) {
-        if (++i == l->max_size) {
-            i = 0;
-        }
+    size_t i;
+    // Do we need to check for this integer overflow ?
```

Comment:
not sure actually, size_t is 32bit on 32bit machines. To get to 4 billion transactions (the only place this code is used) we'd need a lot of traffic. But then again it's probably not completely impossible. From a quick look at the rest of the list code I don't get the impression a overflow is really handled in other places. Would you agree?

Reply:
Indeed, it is not checked in `htp_list_array_get` where there is the same addition `l->first + idx`